lib/Doctrine/KeyValueStore/Storage/AmazonDynamoDbStorage.php

```diff
@@ -0,0 +1,311 @@
+            $keyName = $this->getKeyNameForTable($storageName);
+        }
+
+        return $this->marshaler->marshalItem([$keyName => $keyValue]);
+    }
+
+    /**
+     * Determine if the storage supports updating only a subset of properties,
+     * or if all properties have to be set, even if only a subset of properties
+     * changed.
+     *
+     * @return bool
+     */
+    public function supportsPartialUpdates()
+    {
+        //This is not true, but partial updates are too complicated given the available interface,
+        //meaning, the abstraction is insufficiently flexible enough to support this type of action.
+        return false;
+    }
+
+    /**
+     * Does this storage support composite primary keys?
+     *
+     * @return bool
+     */
+    public function supportsCompositePrimaryKeys()
+    {
+        return true;
+    }
+
+    /**
+     * Does this storage require composite primary keys?
+     *
+     * @return bool
+     */
+    public function requiresCompositePrimaryKeys()
+    {
+        return false;
+    }
+
+    /**
+     * Insert data into the storage key specified.
+     *
+     * @param string       $storageName
+     * @param array|string $key
+     * @param array        $data
+     */
+    public function insert($storageName, $key, array $data)
+    {
+        $this->client->putItem([
+            'TableName' => $storageName,
+            'Item' => $this->prepareKey($storageName, $key) + $this->marshaler->marshalItem($data),
+        ]);
+    }
+
+    /**
+     * Update data into the given key.
+     *
+     * @param string       $storageName
+     * @param array|string $key
+     * @param array        $data
+     */
+    public function update($storageName, $key, array $data)
+    {
+        //We are using PUT so we just replace the original item
+        $this->insert($storageName, $key, $data);
+    }
+
+    /**
+     * Delete data at key.
+     *
+     * @param string       $storageName
+     * @param array|string $key
+     */
+    public function delete($storageName, $key)
+    {
+        $this->client->deleteItem([
+            'TableName' => $storageName,
+            'Key' => $this->prepareKey($storageName, $key),
+        ]);
+    }
+
+    /**
+     * Find data at key.
+     *
+     * Important note: The returned array does contain the identifier (again)!
+     *
+     * @throws NotFoundException When data with key is not found.
+     *
+     * @param string       $storageName
+     * @param array|string $key
+     *
+     * @return array
+     */
+    public function find($storageName, $key)
+    {
+        $item = $this->client->getItem([
+            'TableName' => $storageName,
+            'ConsistentRead' => true,
+            'Key' => $this->prepareKey($storageName, $key),
+        ]);
+
+        if (!$item) {
+            throw new NotFoundException();
+        }
+
+        $item = $item->get('Item');
+        if (!is_array($item)) {
+            throw new NotFoundException();
```

Comment:
Actually, I don't even know if this is possible. I can't find whether or not this type of response is possible from AWS for DynamoDb, so I'm inclined to just remove this check. Thoughts?